rl_zoo3/tracking/tracking_backend.py

```diff
@@ -0,0 +1,124 @@
+        "version/sb3_contrib": f"v{sb3_contrib.__version__}",
+        "version/rl_zoo3": f"v{rl_zoo3.__version__}",
+        "version/docker_image_hash": os.environ["DOCKER_IMAGE_HASH"],
+    }
+
+    def __init__(self):
+        self.initialized = False
+
+    def __init_subclass__(cls, **kwargs) -> None:
+        super().__init_subclass__(**kwargs)
+        short_name = cls.__name__[: -len("Backend")].lower()
+        TrackingBackend.implemented_backends[short_name] = cls
+
+    @classmethod
+    def get_tracker(
+        cls,
+        backend: str,
+    ) -> "TrackingBackend":
+        bd_str = backend.lower()
+        try:
+            return TrackingBackend.implemented_backends[bd_str]()
+        except KeyError:
+            info = ""
+            if bd_str == "wandb":
+                info = "If you want to use Weights & Biases to track experiment, please install W&B via `pip install wandb`."
+            elif bd_str == "mlflow":
+                info = "If you want to use MLflow to track experiment, please install it via `pip install mlflow`."
+            raise KeyError(f"There is no registered backend '{bd_str}'. {info}")
+
+    @classmethod
+    def list_backends(cls) -> Set:
+        return set([k for k in TrackingBackend.implemented_backends.keys()])
+
+    @classmethod
+    @abstractmethod
+    def argparse_add_arguments(cls, parser: argparse.ArgumentParser) -> None:
+        raise NotImplementedError()
+
+    @classmethod
+    @abstractmethod
+    def argparse_del_arguments(cls, parsed_args) -> None:
+        raise NotImplementedError()
+
+    def is_initalized(self) -> bool:
+        return self.initialized
+
+    def get_tracking_commit_hashes(self) -> Dict[str, str]:
+        return TrackingBackend.tracking_commit_hashes
+
+    def get_version_tags(self) -> Dict[str, str]:
+        return TrackingBackend.version_tags
+
+    def setup_tracking(self, args) -> None:
+        if self.is_initalized():
+            return
+        self._setup_tracking(args=args)
+        self.initialized = True
+
+    def finish_tracking(self) -> None:
+        if not self.is_initalized():
+            return
+        self._finish_tracking()
+        self.initialized = False
+
+    def get_sb3_logger(self, verbose: bool) -> Logger | None:
+        if not self.is_initalized():
+            print(
+                f"WARNING: Tried to obtain SB3 Logger without initialization of the tracking backend! Falling back to defaults."
+            )
+            return None
+        return self._get_sb3_logger(verbose=verbose)
+
+    def log_params(self, params: Dict | Any) -> None:
+        if not self.is_initalized():
+            return
+
+        params_dict = params
+        if not isinstance(params, dict):
+            params_dict = vars(params)
+
+        self._log_params(params=params_dict)
+
+    def log_directory(self, local_dir: str, artifacts_dir: str = None) -> None:
+        if not self.is_initalized():
+            return
```

Comment:
is no error/exception/log "cannot log because not initialised..." intended?

Reply:
Added more verbose information and changed `print` functions in favor to the `logging` library.